### PR TITLE
Merge branch 'ganeshnj/feat/otel-tracer' into develop

### PR DESCRIPTION
**Pull request by ganeshnj**
### What and why?

A short description of what changes this PR introduces and why.

### How?

A brief description of implementation details of this PR.

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [x] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [x] Run unit tests for Core, RUM, Trace, Logs, CR and WVT
- [x] Run unit tests for Session Replay
- [x] Run integration tests
- [x] Run smoke tests
- [x] Run tests for `tools/`

### CODE REVIEW

File: DatadogTrace/Sources/Trace.swift [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 import Foundation
 import DatadogInternal
+import OpenTelemetryApi
```
Comment (maciejburda) [inline DatadogTrace/Sources/Trace.swift:9]: not used import? 🤔

File: DatadogTrace/Tests/TraceTests.swift [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 import XCTest
 import TestUtilities
+import OpenTelemetryApi
```
Comment (maciejburda) [inline DatadogTrace/Tests/TraceTests.swift:9]: unused import?

File: Makefile [hunk 3]
```diff
@@ -1,16 +1,6 @@
...
-DD_SDK_SWIFT_TESTING_VERSION = 2.3.2
-
 define DD_SDK_TESTING_XCCONFIG_CI
-DD_SDK_TESTING_PATH=$$(DD_SDK_TESTING_OVERRIDE_PATH:default=$$(SRCROOT)/../instrumented-tests/)\n
-FRAMEWORK_SEARCH_PATHS[sdk=iphonesimulator*]=$$(inherited) $$(DD_SDK_TESTING_PATH)/DatadogSDKTesting.xcframework/ios-arm64_x86_64-simulator/\n
-LD_RUNPATH_SEARCH_PATHS[sdk=iphonesimulator*]=$$(inherited) $$(DD_SDK_TESTING_PATH)/DatadogSDKTesting.xcframework/ios-arm64_x86_64-simulator/\n
-FRAMEWORK_SEARCH_PATHS[sdk=appletvsimulator*]=$$(inherited) $$(DD_SDK_TESTING_PATH)/DatadogSDKTesting.xcframework/tvos-arm64_x86_64-simulator/\n
-LD_RUNPATH_SEARCH_PATHS[sdk=appletvsimulator*]=$$(inherited) $$(DD_SDK_TESTING_PATH)/DatadogSDKTesting.xcframework/tvos-arm64_x86_64-simulator/\n
-OTHER_LDFLAGS[sdk=iphonesimulator*]=$$(inherited) -framework DatadogSDKTesting\n
-OTHER_LDFLAGS[sdk=appletvsimulator*]=$$(inherited) -framework DatadogSDKTesting\n
```
Comment (maciejburda) [inline Makefile:13]: I guess we should revert this and bump the Testing SDK version?

**Review by maciejburda — COMMENTED** [review]
Woah! Big big effort! 👏 💪  

Gave it a pass and left some minor comments. Let me know if they make sense.

Since it was dogfooded, it seems like we're good to go. Wondering if it would make sense to fix flaky CI before we merge this.

File: DatadogCore/Tests/Datadog/DatadogCore/DatadogCoreTests.swift [hunk 4]
```diff
@@ -146,7 +146,6 @@ class DatadogCoreTests: XCTestCase {
             maxBatchesPerUpload: .mockRandom(min: 1, max: 100),
             backgroundTasksEnabled: .mockAny()
         )
-        defer { core.flushAndTearDown() }
```
[the remark this answers is not in the document]
Comment (ganeshnj) [reply DatadogCore/Tests/Datadog/DatadogCore/DatadogCoreTests.swift]: it looks like a bad merge.

File: DatadogTrace/Tests/TraceTests.swift (hunk 2 above)
Comment (ganeshnj) [reply DatadogTrace/Tests/TraceTests.swift]: Indeed.

File: Makefile (hunk 3 above)
Comment (ganeshnj) [reply Makefile:13]: Nope, we no longer use the testing SDK as a dynamically dependency. It is now integrated as SPM package.

File: Datadog/Datadog.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -13332,6 +13489,30 @@
...
+			isa = XCSwiftPackageProductDependency;
+			package = 3CDA3F6C2BCD8429005D2C13 /* XCRemoteSwiftPackageReference "dd-sdk-swift-testing" */;
+			productName = DatadogSDKTesting;
+		};
+		3CDA3F7F2BCD8687005D2C13 /* DatadogSDKTesting */ = {
+			isa = XCSwiftPackageProductDependency;
+			package = 3CDA3F6C2BCD8429005D2C13 /* XCRemoteSwiftPackageReference "dd-sdk-swift-testing" */;
+			productName = DatadogSDKTesting;
+		};
+/* End XCSwiftPackageProductDependency section */
```
[the remark this answers is not in the document]
Comment (ganeshnj) [reply Datadog/Datadog.xcodeproj/project.pbxproj:13513]: No, this is expected, we changed the way testing SDK is used in our CI. It is not consumed via SPM.

**Pull request merged**